cassandra-sink/src/main/java/org/springframework/xd/module/options/mixin/CassandraConnectionMixin.java

```diff
@@ -0,0 +1,151 @@
+
+import org.hibernate.validator.constraints.NotBlank;
+import org.hibernate.validator.constraints.Range;
+
+/**
+ * @author Artem Bilan
+ */
+public class CassandraConnectionMixin {
+
+	private String contactPoints = CassandraCqlClusterFactoryBean.DEFAULT_CONTACT_POINTS;
+
+	private int port = CassandraCqlClusterFactoryBean.DEFAULT_PORT;
+
+	private String keyspace = ModulePlaceholders.XD_STREAM_NAME;
+
+	private String username;
+
+	private String password;
+
+	private SchemaAction schemaAction = SchemaAction.NONE;
+
+	private String[] entityBasePackages = new String[0];
+
+	private CompressionType compressionType = CompressionType.NONE;
+
+	private boolean metricsEnabled = CassandraCqlClusterFactoryBean.DEFAULT_METRICS_ENABLED;
+
+	@ModuleOption("the comma-delimited string of the hosts to connect to Cassandra")
+	public void setContactPoints(String contactPoints) {
+		this.contactPoints = contactPoints;
+	}
+
+	@ModuleOption("the port to use to connect to the Cassandra host")
+	public void setPort(int port) {
+		this.port = port;
+	}
+
+	@ModuleOption("the keyspace name to connect to")
+	public void setKeyspace(String keyspace) {
+		this.keyspace = keyspace;
+	}
+
+	@ModuleOption("the username for connection")
+	public void setUsername(String username) {
+		this.username = username;
+	}
+
+	@ModuleOption("the password for connection")
+	public void setPassword(String password) {
+		this.password = password;
+	}
+
+	@ModuleOption("the Cassandra schema action")
+	public void setSchemaAction(SchemaAction schemaAction) {
+		this.schemaAction = schemaAction;
+	}
+
+	@ModuleOption("the base packages to scan for entities annotated with Table annotations")
+	public void setEntityBasePackages(String[] entityBasePackages) {
+		this.entityBasePackages = entityBasePackages;
+	}
+
+	@ModuleOption("the compression to use for the transport")
+	public void setCompressionType(CompressionType compressionType) {
+		this.compressionType = compressionType;
+	}
+
+	@ModuleOption("enable/disable metrics collection for the created cluster")
+	public void setMetricsEnabled(boolean metricsEnabled) {
+		this.metricsEnabled = metricsEnabled;
+	}
+
+	@NotBlank
+	public String getContactPoints() {
+		return this.contactPoints;
+	}
+
+	@Range(min = 0, max = 65535)
+	public int getPort() {
+		return this.port;
+	}
+
+	public String getKeyspace() {
+		return this.keyspace;
+	}
+
+	public String getUsername() {
+		return this.username;
+	}
+
+	public String getPassword() {
+		return this.password;
+	}
+
+	@NotNull
+	public SchemaAction getSchemaAction() {
+		return schemaAction;
+	}
+
+	public String[] getEntityBasePackages() {
+		return this.entityBasePackages;
+	}
+
+	@NotNull
+	public CompressionType getCompressionType() {
+		return this.compressionType;
+	}
+
+	public boolean isMetricsEnabled() {
+		return this.metricsEnabled;
+	}
+
+	@AssertFalse(message = "both 'username' and 'password' are required or no one")
```

Comment:
As when a violation occurs the bean "property" name is displayed, it's better to turn this around and have it named "isInvalid" or something. That way, the user sees:
```
invalid: both 'username' and 'password' are required or no one
```
rather than
```
valid: both 'username' and 'password' are required or no one
```